demos/interactive_face_detection_demo_gapi/main.cpp

```diff
@@ -0,0 +1,573 @@
+}
+
+void faceDataUpdate(const cv::Mat &frame,
+                    Face::Ptr &face,
+                    const cv::Rect &face_rect,
+                    std::list<Face::Ptr>  &prev_faces,
+                    const std::vector<cv::Rect> &face_hub,
+                    size_t &id,
+                    bool no_smooth) {
+    // Face update
+    cv::Rect rect = face_rect & cv::Rect({0, 0}, frame.size());
+
+    if (!no_smooth) {
+        face = matchFace(rect, prev_faces);
+        float intensity_mean = calcMean(frame(rect));
+        intensity_mean += 1.0;
+
+        if ((face == nullptr) ||
+            ((std::abs(intensity_mean - face->_intensity_mean) / face->_intensity_mean) > 0.07f)) {
+            face = std::make_shared<Face>(id++, rect);
+        } else {
+            prev_faces.remove(face);
+        }
+
+        face->_intensity_mean = intensity_mean;
+        face->_location = rect;
+    } else {
+        face = std::make_shared<Face>(id++, rect);
+    }
+}
+
+void ageGenderDataUpdate(const Face::Ptr &face,
+                         const cv::Mat &out_age,
+                         const cv::Mat &out_gender) {
+    const float *age_data =    out_age.ptr<float>();
+    const float *gender_data = out_gender.ptr<float>();
+
+    float maleProb = gender_data[1];
+    float age      = age_data[0] * 100;
+
+    face->updateGender(maleProb);
+    face->updateAge(age);
+}
+
+void headPoseDataUpdate(const Face::Ptr &face,
+                        const cv::Mat &out_y_fc,
+                        const cv::Mat &out_p_fc,
+                        const cv::Mat &out_r_fc) {
+    const float *y_data = out_y_fc.ptr<float>();
+    const float *p_data = out_p_fc.ptr<float>();
+    const float *r_data = out_r_fc.ptr<float>();
+
+    face->updateHeadPose(y_data[0], p_data[0], r_data[0]);
+}
+
+void emotionsDataUpdate(const Face::Ptr &face, const cv::Mat &out_emotion) {
+    const float *em_data = out_emotion.ptr<float>();
+
+    std::map<std::string, float> em_val_map;
+    for(size_t i = 0; i  < EMOTION_VECTOR.size(); i++) {
+        em_val_map[EMOTION_VECTOR[i]] = em_data[i];
+    }
+
+    face->updateEmotions(em_val_map);
+}
+
+void landmarksDataUpdate(const Face::Ptr &face, const cv::Mat &out_landmark) {
+    const float *lm_data = out_landmark.ptr<float>();
+
+    size_t n_lm = 70;
+
+    std::vector<float> normedLandmarks(&lm_data[0], &lm_data[n_lm]);
+
+    face->updateLandmarks(normedLandmarks);
+}
+
+void setInput(cv::GStreamingCompiled stream, const std::string& input ) {
+    try {
+        // If stoi() throws exception input should be a path not a camera id
+        stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(std::stoi(input)));
+    } catch (std::invalid_argument&) {
+        slog::info << "Input source is treated as a file path" << slog::endl;
+        stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(input));
+    }
+}
+
+// Copied from demos/common/include/samples/common.hpp
```

Comment:
This comment isn't very helpful, because the function no longer exists in that file. Just remove it.